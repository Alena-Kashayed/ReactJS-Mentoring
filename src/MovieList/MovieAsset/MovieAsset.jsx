import React from 'react';
import styles from './MovieAsset.scss';

function MovieAsset(props) {
    const showDetails = () => {
        props.showDetails(props);
    };

    return (
        <div className={styles.movieAsset}>
            <img className={styles.moviePoster} src={props.url} alt={props.name} onClick={showDetails}/>
            <div className={styles.movieTitle}>
                <div>{props.name}</div>
                <div className={styles.movieYear}>{props.year}</div>
            </div>
            <div className={styles.movieGenre}>{props.genre}</div>
        </div>
    );
}

export default MovieAsset;
