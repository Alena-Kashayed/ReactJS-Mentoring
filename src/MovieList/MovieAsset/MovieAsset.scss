.movieAsset {
  flex-basis: 33%;
  text-align: center;
  margin-bottom: 35px;

  .poster {
    cursor: pointer;
  }

  .moviePoster {
    width: 240px;
    height: 330px;
  }

  .movieTitle {
    display: flex;
    justify-content: space-between;
    padding: 5px;
    width: 240px;
    margin: 0 auto;
    box-sizing: border-box;

    .movieYear {
      border: 1px solid #525252;
      padding: 3px;
      border-radius: 3px;
      font-size: 12px;
    }
  }

  .movieGenre {
    font-size: 12px;
    padding: 5px;
    width: 240px;
    text-align: left;
    margin: 0 auto;
    box-sizing: border-box;
  }
}
