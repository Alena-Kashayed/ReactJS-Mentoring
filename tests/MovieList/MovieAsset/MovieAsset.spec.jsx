import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import MovieAsset from '../../../src/MovieList/MovieAsset/MovieAsset';

describe('MovieAsset', () => {
  it('is rendered correctly', () => {
    const asset = {
      id: 1,
      title: 'Title',
      release_date: '2017-01-01',
      first_air_date: '2017-01-01',
      poster_path: '/123.jpg',
    };
    const typeOfQuery = 'movie';
    const wrapper = shallow(<MovieAsset
      asset={asset}
      typeOfQuery={typeOfQuery}
    />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
