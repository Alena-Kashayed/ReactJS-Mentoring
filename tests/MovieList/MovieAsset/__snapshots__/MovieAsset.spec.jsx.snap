// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieAsset is rendered correctly 1`] = `
<div
  className="movieAsset"
>
  <Link
    replace={false}
    to="/film/movie/1"
  >
    <div
      className="poster"
      role="presentation"
    >
      <img
        alt="Title"
        className="moviePoster"
        src="https://image.tmdb.org/t/p/w300/123.jpg"
      />
    </div>
  </Link>
  <div
    className="movieTitle"
  >
    <div>
      Title
    </div>
    <div
      className="movieYear"
    >
      2017
    </div>
    <div
      className="movieYear"
    >
      2017
    </div>
  </div>
</div>
`;
